Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,136 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using Microsoft.SqlServer.Server;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Model.Building;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Parameters;
+using Remotion.Data.DomainObjects.UnitTests.Factories;
+using Remotion.Data.DomainObjects.UnitTests.TestDomain;
+using Remotion.Development.NUnit.UnitTesting;
+using Remotion.Development.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SqlServer.Parameters;
+
+[TestFixture]
+public class SqlTableValuedDataParameterDefinitionTest
+{
+  private static IEnumerable<(object[], bool)> GetTestCollections ()
+  {
+    yield return ([1, 2, 3, 5, 8, 13, 21], true); // distinct
+    yield return ([1, 2, 3, 3, 2, 1], false); // duplicates
+    yield return ([], false); // empty
+    yield return ([], true); // empty distinct
+    yield return ([ClassWithAllDataTypes.EnumType.Value0, ClassWithAllDataTypes.EnumType.Value1, ClassWithAllDataTypes.EnumType.Value2], true); // converted
+    yield return (["Value1", "Value2", "Value1"], false);
+  }
+
+  [Test]
+  [TestCaseSource(nameof(GetTestCollections))]
+  public void GetParameterValue_ReturnsSqlTableValuedParameterValue ((object[], bool) testCase)
+  {
+    var items = testCase.Item1;
+    var isDistinct = testCase.Item2;
+
+    var storageTypeInformation = new SqlStorageTypeInformationProvider().GetStorageType(items.FirstOrDefault());
+    var convertedValues = items.Select(i => storageTypeInformation.ConvertToStorageType(i)).ToArray();
+
+    var parameterDefinition = new SqlTableValuedDataParameterDefinition(storageTypeInformation, isDistinct);
+
+    var result = parameterDefinition.GetParameterValue(items);
+
+    Assert.That(result, Is.InstanceOf<SqlTableValuedParameterValue>());
+
+    var tvpValue = result.As<SqlTableValuedParameterValue>();
+
+    if(isDistinct)
+      Assert.That(tvpValue.TableTypeName, Is.EqualTo($"TVP_{storageTypeInformation.StorageDbType}{SqlTableValuedDataParameterDefinition.DistinctCollectionTableTypeNameSuffix}"));
+    else
+      Assert.That(tvpValue.TableTypeName, Is.EqualTo($"TVP_{storageTypeInformation.StorageDbType}"));
+
+    Assert.That(tvpValue.ColumnMetaData.Count, Is.EqualTo(1));
+
+    var sqlMetaData = tvpValue.ColumnMetaData.Single();
+    Assert.That(sqlMetaData.Name, Is.EqualTo("Value"));
+
+    var conversionHelper = new SqlParameter();
+    conversionHelper.DbType = storageTypeInformation.StorageDbType;
+    var sqlDbType = conversionHelper.SqlDbType;
```

Comment:
I would prefer if we had this as part of the testset instead of calculated. the fewer computations a test has to do, the less likely a bug is introduced. also, it documents what is expected. With the current test, if there's a bug with using SqlParamerter (e.g. it always does DbType.Object or something) we might not find it if the prod-code uses the same apis.